flink-core/src/main/java/org/apache/flink/api/connector/sink/SinkWriter.java

```diff
@@ -53,7 +53,7 @@
      * @param watermark The watermark.
      * @throws IOException if fail to add a watermark.
      */
-    default void processWatermark(Watermark watermark) throws IOException {}
+    default void writeWatermark(Watermark watermark) throws IOException {}
```

Comment:
Yes this name feels better when looking at your tests. Also agree that it's better in `SinkFunction`.